### ISSUE TITLE
Can't change `Select` value programmatically

### ISSUE THREAD

**Issue by mattiaz9** [daisyui/react-daisyui#236]
Since the value of the select is stored in a local state of the Select component, any manual change of the value don't update the selected option.

Possbile fix (default react behavior):
1. Directly pass the `value` to the `<select>` tag
2. Rename `initialValue` in `defaultValue` and pass it directly in the html tag as well

So basically the code should look like this:
```ts
const SelectInner = <T extends string | number | undefined>(
  props: SelectProps<T>,
  ref: React.ForwardedRef<HTMLSelectElement>
): JSX.Element => {
  const {
    children,
    defaultValue,
    value,
    onChange,
    size,
    color,
    bordered = true,
    borderOffset,
    dataTheme,
    className,
    ...rest
  } = props

  const classes = twMerge(
    'select',
    className,
    clsx({
      [`select-${size}`]: size,
      [`select-${color}`]: color,
      [`focus:outline-offset-0`]: !borderOffset,
      'select-bordered': bordered,
    })
  )

  return (
    <select
      {...rest}
      ref={ref}
      data-theme={dataTheme}
      className={classes}
      onChange={(e) => onChange?.(e.currentTarget.value as T)}
      value={value}
      defaultValue ={defaultValue}
    >
      {children}
    </select>
  )
}
```

**Comment by Klaas058** [comment]
Hi @mattiaz9 thank you for opening the issue!

I agree that you should be able change the value programmatically, though a combination of `value` and `defaultValue` would not be inline with the React guidelines.
A component should either be a [controlled](https://reactjs.org/docs/forms.html#controlled-components) component or an [uncontrolled](https://reactjs.org/docs/uncontrolled-components.html) component.
The `value` prop is for `controlled` components and the `defaultValue` is for `uncontrolled` components.

I would propose to have a useEffect hook to listen to changes to the `value` prop and changes the selectedValue when triggered.

What do you think?

**Comment by mattiaz9** [comment]
@Klaas058 if you think it is better I'm not opposed to that as long as it works.

Personally though I don't like it very much. The `Select` component should be an extension of the `<select />` html tag and if you were to use a normal select you either use a value/onChange controlled approach or you use the defaultValue approach.
Either way is the developer that should deal with this problem, not the library IMHO.

**Comment by Klaas058** [comment]
@mattiaz9 I see what you mean, I'm not opposed making the component in the **library** an `uncontrolled` component so that the control is left to the developer.

I would be happy with your proposal though I don't see a use-case for the `defaultValue`.
Since `defaultValue` is already in the HTMLSelectElement props we don't need it in ours since we don't change anything.
We can just leave it to the `{...rest}` splitting.

I think you are right about the library statement.

If you make a pull request for this change and it is merged, then I will also reflect that philosophy in the `Rating` component.
Which I wrote with internal state which, now that we talked about it, would not be necessary or fitting.
